### PR TITLE
Add a self update command to update the phar file.

### LINKED ISSUE #102: Phar self update
There aren't a lot of update for now (personally I'm focusing on releasing PHP-DI right now) but it would be good to add an auto-update system (e.g. `couscous selfupdate`).

It seems really easy to do with this package: https://github.com/padraic/phar-updater

### PR DESCRIPTION
**Pull request by wysow**
Fixes #102

@mnapoli for this to work we first need to push a new couscous.phar and couscous.version online, can I do it?

### CODE REVIEW

File: src/Application/Cli/SelfUpdateCommand.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        try {
+            $updater = new Updater(null, false);
+            $updater->getStrategy()->setPharUrl('http://couscous.io/couscous.phar');
+            $updater->getStrategy()->setVersionUrl('http://couscous.io/couscous.version');
+
+            $result = $updater->update();
+            $result ? $output->writeln('Phar file updated successfully!') : $output->writeln('No need to update.');
+        } catch (\Exception $e) {
+            die(var_dump($e->getMessage()));
```
Comment (mnapoli) [inline src/Application/Cli/SelfUpdateCommand.php]: `die()` here :)

File: bin/compile [hunk 2]
```diff
@@ -58,3 +58,5 @@ $builder->setStub(
 );
 
 chmod(__DIR__ . '/couscous.phar', 0755);
+
+exec('sha1sum bin/couscous.phar > bin/couscous.version');
```
Comment (mnapoli) [inline bin/compile]: Could we use something more widespread? I don't have that installed, I have `md5` though. We can also do it with PHP (`php -r "..."`) if needed.

**Comment by mnapoli** [comment]
:+1: feel free to push a new version online.

**Comment by wysow** [comment]
@mnapoli This should be good to go with this one...

**Comment by mnapoli** [comment]
Merging!

**Pull request merged**